rails_programming/forms_and_authentication/form_basics.md

```diff
@@ -126,24 +126,38 @@ Start by making a form using the `form_tag` helper, which takes a block represen
 Creates the form:
 
 ~~~html
-  <form accept-charset="UTF-8" action="/search" method="get">
+  <form accept-charset="UTF-8" action="/search" data-remote="true" method="get">
     <label for="q">Search for:</label>
     <input id="q" name="q" type="text" />
-    <input name="commit" type="submit" value="Search" />
+    <input name="commit" type="submit" value="Search" data-disable-with="Search" />
   </form>
 ~~~
 
-The ID of the inputs matches the name.
-
 There are tag helpers for all the major tags and the options they accept are all a bit different.  See the reading assignment for more detail.
 
-### Handy Shortcuts: `form_for`
+There are a few things to take note of when using the `form_with` helper.
+
+1. The ID of the inputs matches the name.
+
+2. The second line ends with `as JS` instead of the usual `as HTML` when you look at your output in your console after submitting a form. By default, all forms using form_with will submit data using an XHR (Ajax) request. To disable this include `local: true` when building your form like this.
+
+~~~bash
+  <%= form_with(url: "/search", method: "get"), local: true do %>
+    <%= label_tag(:q, "Search for:") %>
+    <%= text_field_tag(:q) %>
+    <%= submit_tag("Search") %>
+  <% end %>
+~~~
+
+You can also check your Network tab in your browser to see the requests in both cases.
 
-No one wants to remember to specify which URL the form should submit to or write out a whole bunch of `*_tag` methods, so Rails gives you a shortcut in the form of the slightly more abstracted `form_for` method.  It's a whole lot like `form_tag` but does a bit more work for you.
+### Using models with the `form_with` helper
 
-Just pass `form_for` a model object, and it will make the form submit to the URL for that object, e.g. `@user` will submit to the correct URL for creating a User.  Remember from the lesson on controllers that the `#new` action usually involves creating a new (unsaved) instance of your object and passing it to the view... now you finally get to see why by using that object in your `#form_for` forms!
+No one wants to remember to specify which URL the form should submit to or write out a whole bunch of `*_tag` methods. Using the `form_with` helper with a model can make things easy and quick.
 
-Where `form_tag` accepted a block without any arguments and the individual inputs had to be specified with `something_tag` syntax, `form_for` actually passes the block a form object and then you create the form fields based off that object.  It's conventional to call the argument simply `f`.
+Just pass `form_with` a model object, and it will make the form submit to the URL for that object, e.g. `@user` will submit to the correct URL for creating a User.  Remember from the lesson on controllers that the `#new` action usually involves creating a new (unsaved) instance of your object and passing it to the view... now you finally get to see why by using that object in your `#form_with` forms!
```

Comment:
lets stick with using `@article` as the example vs @user given the snippet below is for an Article model for consistency.